src/main/webapp/app/exam/participate/exercises/modeling/modeling-exam-submission.component.ts

```diff
@@ -116,4 +117,29 @@ export class ModelingExamSubmissionComponent extends ExamSubmissionComponent imp
         this.studentSubmission.isSynced = false;
         this.explanationText = explanation;
     }
+
+    async setSubmissionVersion(submission: SubmissionVersion): Promise<void> {
+        this.submissionVersion = submission;
+        await this.updateViewFromSubmissionVersion();
+    }
+
+    private async updateViewFromSubmissionVersion() {
+        if (this.submissionVersion) {
+            if (this.submissionVersion.content) {
+                // we need these string operations because we store the string like that in the database
+                // and need to remove the content that was added before the string is saved to the db to get valid JSON
+                let model = this.submissionVersion.content.substring(0, this.submissionVersion.content.indexOf('; Explanation:'));
+                // if we do not wait here for apollon, the redux store might be undefined
+                await this.modelingEditor!.apollonEditor!.nextRender;
+                model = model.replace('Model: ', '');
+                // Updates the Apollon editor model state (view) with the latest modeling submission
+                this.umlModel = JSON.parse(model);
+                // same as above regarding the string operations
+                this.explanationText = this.submissionVersion.content.substring(this.submissionVersion.content.indexOf('Explanation:') + 13) ?? '';
```

Comment:
Why `+13`? "explanation:" has 12 characters 🤔 A comment about this or some constant would be helpful